src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1381,6 +1381,13 @@ public function __call($method, $parameters)
             return $this->toBase()->{$method}(...$parameters);
         }
 
+        // Ignore calls to the underlying QueryBuilder's `asClass`/`as` method
+        // as Eloquent models already are converted into an object.
+        // We will silently ignore this call, as it's not an exceptional case.
+        if (in_array($method, ['as', 'asClass'])) {
```

Comment:
The third parameter for strict type comparison should be used to avoid implicit type comparisons.